curp/tests/server.rs

```diff
@@ -379,3 +379,64 @@ async fn propose_remove_node_failed() {
     let res = client.propose_conf_change(conf_change).await.unwrap();
     assert!(matches!(res, Err(ConfChangeError::InvalidConfig(()))));
 }
+
+#[tokio::test(flavor = "multi_thread")]
+#[abort_on_panic]
+async fn shutdown_rpc_should_shutdown_the_cluster_when_client_has_wrong_leader() {
+    init_logger();
+    let tmp_path = tempfile::TempDir::new().unwrap().into_path();
+    let group = CurpGroup::new_rocks(3, tmp_path.clone()).await;
+
+    let leader_id = group.get_leader().await.0;
+    let follower_id = group
+        .all
+        .keys()
+        .find(|&id| &leader_id != id)
+        .copied()
+        .unwrap();
+    // build a client and set a wrong leader id
+    let client = Client::<TestCommand>::builder()
+        .config(ClientConfig::default())
+        .build_from_all_members(group.all(), Some(follower_id))
+        .await
+        .unwrap();
+
+    client.shutdown().await.unwrap();
+
+    sleep_secs(3).await; // wait for the cluster to shutdown
+    assert!(group.is_finished());
+}
+
+#[tokio::test(flavor = "multi_thread")]
+#[abort_on_panic]
+async fn propose_conf_change_to_learner() {
+    init_logger();
+    let group = CurpGroup::new(5).await;
+
+    let leader_id = group.get_leader().await.0;
+    let follower_id = group
+        .all
+        .keys()
+        .find(|&id| &leader_id != id)
+        .copied()
+        .unwrap();
+    // build a client and set a wrong leader id
+    let client = Client::<TestCommand>::builder()
+        .config(ClientConfig::default())
+        .build_from_all_members(group.all(), Some(follower_id))
+        .await
+        .unwrap();
+
+    let id = generate_propose_id("test");
+    let node_id = group.nodes.keys().next().copied().unwrap();
+    let changes = vec![ConfChange::update(node_id, "new_addr".to_owned())];
+    let conf_change = ProposeConfChangeRequest::new(id, changes);
+    let members = client
```

Comment:
It seems you propose the `ProposeConfChangeRequest` to a follower, not a learner. Is it correct?

Reply:
the test name should be `follower`, Fixed